pom/homepage.py

```diff
@@ -1,29 +1,110 @@
-
-    # Methods for getting the text of each title button
-    def get_nav_links_text(self) -> List[str]:
-        nav_links = self.get_nav_links()
-        nav_links_text = [link.text for link in nav_links]
-        return ",".join(nav_links_text).split('\n')
-
-    # Method for getting the text of Build & Price button
-    def get_nav_link_text(self) -> str:
-        nav_link_text = self.get_category().text
-        return nav_link_text
+    button_crossovers_and_SUVs = '.c_310_tertiary_grid_slider button[data-bodystyle="Crossovers & SUVs"] .c_310-vehicles-count'
+    cars_in_crossovers_and_SUVs = '#vehiclelist--1738608765_categoryPanel_0 .c_310-1'
+
+    click_cars = '.c_310_tertiary_grid_slider [data-label="Cars"]'
+    button_cars = '.c_310_tertiary_grid_slider button[data-bodystyle="Cars"] .c_310-vehicles-count'
+    cars_in_cars = '#vehiclelist--1738608765_categoryPanel_1 .c_310-1'
+
+    click_electic = '.c_310_tertiary_grid_slider button[data-bodystyle="Electric"]'
+    button_electric = '.c_310_tertiary_grid_slider button[data-bodystyle="Electric"] .c_310-vehicles-count'
+    cars_in_electric = '#vehiclelist--1738608765_categoryPanel_2 .c_310-1'
+
+    click_sports_cars = '.c_310_tertiary_grid_slider button[data-bodystyle="Sports Cars"]'
+    button_sports_cars = '.c_310_tertiary_grid_slider button[data-bodystyle="Sports Cars"] .c_310-vehicles-count'
+    cars_in_sports_cars = '#vehiclelist--1738608765_categoryPanel_3 .c_310-1'
+
+    click_trucks = '.c_310_tertiary_grid_slider button[data-bodystyle="Trucks"]'
+    button_trucks = '.c_310_tertiary_grid_slider button[data-bodystyle="Trucks"] .c_310-vehicles-count'
+    cars_in_trucks = '#vehiclelist--1738608765_categoryPanel_4 .c_310-1'
+
+    '''Crossovers & SUVs'''
+
+    # Return a Crossovers & SUVs tab
+    def get_category_crossovers_and_SUVs(self) -> WebElement:
+        return self.is_visible('css', self.button_crossovers_and_SUVs, 'Crossovers & SUVs')
+
+    # Method for getting the text of Crossovers & SUVs tab
+    def get_category_crossovers_and_SUVs_text(self):
+        return self.get_category_crossovers_and_SUVs().text
+
+    # Return number of cars displayed on Crossovers & SUVs grid pannel
+    def get_num_crossovers_and_SUVs(self) -> int:
+        return len(self.are_visible('css', self.cars_in_crossovers_and_SUVs, 'Cars in Crossovers & SUVs tab'))
+
+    '''Cars'''
+
+    # Return a tab for click
+    def get_click_cars(self) -> WebElement:
+        return self.is_visible('css', self.click_cars, 'Click Cars')
+
+    # Return a Cars tab
+    def get_category_cars(self) -> WebElement:
+        return self.is_visible('css', self.button_cars, 'Count Cars')
+
+    # Method for getting the text of Cars tab
+    def get_category_cars_text(self):
+        return self.get_category_cars().text
+
+    # Return number of cars displayed on Cars grid pannel
+    def get_num_cars(self) -> int:
+        return len(self.are_visible('css', self.cars_in_cars, 'Cars in Cars tab'))
+
```

Comment:
I suggest to break this method into two logical methods:
1. first method returns a list of web elements: self.are_visible('css', self.cars_in_cars, 'Cars in Cars tab')
2. second method return a length from a list of web elements. Please make this method a general one, so you could use any list of web elements as an argument
def get_num_cars(self, elements: List[WebEelement]) -> int:
    return len(elements)